core/drivers/stm32_gpio.c

```diff
@@ -77,6 +80,132 @@ static unsigned int gpio_lock;
 static STAILQ_HEAD(, stm32_gpio_bank) bank_list =
 		STAILQ_HEAD_INITIALIZER(bank_list);
 
+static struct stm32_gpio_bank *gpio_chip_to_bank(struct gpio_chip *chip)
+{
+	return container_of(chip, struct stm32_gpio_bank, gpio_chip);
+}
+
+static enum gpio_level stm32_gpio_get_level(struct gpio_chip *chip,
+					    unsigned int gpio_pin)
+{
+	struct stm32_gpio_bank *bank = gpio_chip_to_bank(chip);
+	enum gpio_level level = GPIO_LEVEL_HIGH;
+	unsigned int reg_offset = 0;
+	unsigned int mode = 0;
+
+	assert(gpio_pin < bank->ngpios);
```

Comment:
gpio_pin is an argument that is provided by a framework user, therefore is not static like device tree configuration. assert will disappear when debug isn't enabled. Maybe a if/panic is better here.

(several occurences)

Reply:
which framework do you foresee on manipulating GPIOs?
A consumer driver should not get/set a GPIO level if it did not get the GPIO first. So I think the assertion above is even overkilling.

Reply:
Maybe when piloting a GPIO regulator?

Whatever the response, I think your statement : _A consumer driver should not get/set a GPIO level if it did not get the GPIO first. So I think the assertion above is even overkilling._ is valid. It can even be removed